test/cmocka/src/math/arithmetic/exponential.c

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <math.h>
+#include <cmocka.h>
+#include <string.h>
+#include <time.h>
+#include <stdlib.h>
+
+#include <sof/math/exp_fcn.h>
+#include <sof/audio/format.h>
+#include <rtos/string.h>
+#include <sof/common.h>
+
+#define ULP_TOLERANCE 5.60032793
+#define ULP_SCALE 0.0000999999999749
+
+static void gen_exp_testvector(double a, double b, double *r, int32_t *b_i);
+
+/* testvector in Q4.28, -5 to +5 */
+/*
+ * Arguments	:double a
+ *		 double b
+ *		 double *r
+ *		 int32_t *b_i
+ * Return Type	:void
+ */
+static void gen_exp_testvector(double a, double b, double *r, int32_t *b_i)
+{
+	double a2;
+	double b2;
+
+	a2 = a / 2.0;
```

Comment:
As reported when you ran `ccpcheck`, what is the purpose of this line? It has no effect.

Reply:
done!!

The test was rerun, and the cppcheck results are presented below.
I don't notice any prior warnings.
I'm curious if every PR passes the cppcheck, and if not, what criteria is used ?

```
[/sof/test/cmocka/src/math/arithmetic/exponential.c:61] (style) The scope of the variable 'accum' can be reduced. Warning: Be careful when fixing this message, especially when there are inner loops. Here is an example where cppcheck will write that the scope for 'i' can be reduced:
void f(int x)
{
    int i = 0;
    if (x) {
        // it's safe to move 'int i = 0;' here
        for (int n = 0; n < 10; ++n) {
            // it is possible but not safe to move 'int i = 0;' here
            do_something(&i);
        }
    }
}
When you see this message it is always safe to reduce the variable scope 1 level. [variableScope]
[/sof/test/cmocka/src/math/arithmetic/exponential.c:63] (style) The scope of the variable 'a_i' can be reduced. Warning: Be careful when fixing this message, especially when there are inner loops. Here is an example where cppcheck will write that the scope for 'i' can be reduced:
void f(int x)
{
    int i = 0;
    if (x) {
        // it's safe to move 'int i = 0;' here
        for (int n = 0; n < 10; ++n) {
            // it is possible but not safe to move 'int i = 0;' here
            do_something(&i);
        }
    }
}
When you see this message it is always safe to reduce the variable scope 1 level. [variableScope]
[/sof/test/cmocka/src/math/arithmetic/exponential.c:64] (style) The scope of the variable 'max_ulp' can be reduced. Warning: Be careful when fixing this message, especially when there are inner loops. Here is an example where cppcheck will write that the scope for 'i' can be reduced:
void f(int x)
{
    int i = 0;
    if (x) {
        // it's safe to move 'int i = 0;' here
        for (int n = 0; n < 10; ++n) {
            // it is possible but not safe to move 'int i = 0;' here
            do_something(&i);
        }
    }
}
When you see this message it is always safe to reduce the variable scope 1 level. [variableScope]

```